src/coreclr/src/jit/inlinepolicy.cpp

```diff
@@ -461,6 +439,53 @@ void DefaultPolicy::NoteBool(InlineObservation obs, bool value)
     }
 }
 
+//------------------------------------------------------------------------
+// BudgetCheck: see if this inline would exceed the current budget
+//
+// Returns:
+//   True if inline would exceed the budget.
+//
+bool DefaultPolicy::BudgetCheck() const
+{
+    // Only relevant if we're actually inlining.
+    //
+    if (m_IsPrejitRoot)
+    {
+        return false;
+    }
+
+    // The strategy tracks the amout of inlining done so far,
+    // so it performs the actual check.
+    //
+    InlineStrategy* strategy   = m_RootCompiler->m_inlineStrategy;
+    const bool      overBudget = strategy->BudgetCheck(m_CodeSize);
+
+    if (overBudget)
```

Comment:
Nit: If we will have more cases like this it could be confusing.

Reply:
I'm not sure I follow -- does "more cases" refer to more overrides of the `BudgetCheck`, or more reasons we might allow inlines to go over budget?

I will probably adjust this bit of the default policy soon to address #41692.

Reply:
More reasons and more overrides where we allow to go over budget when the header says `True if inline would exceed the budget.` and the user is `const bool overBudget = this->BudgetCheck();`

Reply:
There's only ever going to be one reason to go over budget -- when we have an aggressive inline that we suspect might end up being less costly than it looks.

And likely no more overrides.